customer/src/main/java/no/unit/nva/customer/events/producer/DefaultCustomerResourceUpdateEventsProducer.java

```diff
@@ -0,0 +1,113 @@
+import no.unit.nva.customer.events.model.ChannelClaim;
+import no.unit.nva.customer.events.model.ChannelClaimUpdateEvents;
+import no.unit.nva.customer.events.model.ResourceUpdateEvent;
+import no.unit.nva.customer.model.CustomerDao;
+import no.unit.nva.customer.model.CustomerDto;
+import no.unit.nva.customer.model.channelclaim.ChannelClaimDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+
+public class DefaultCustomerResourceUpdateEventsProducer implements CustomerResourceUpdateEventsProducer {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultCustomerResourceUpdateEventsProducer.class);
+    private static final TableSchema<CustomerDao> CUSTOMER_DAO_TABLE_SCHEMA = TableSchema.fromBean(CustomerDao.class);
+
+    private final AttributeValueConverter attributeValueConverter;
+
+    public DefaultCustomerResourceUpdateEventsProducer(AttributeValueConverter attributeValueConverter) {
+        this.attributeValueConverter = attributeValueConverter;
+    }
+
+    @Override
+    public List<ResourceUpdateEvent<ChannelClaim>> produceEvents(DynamodbStreamRecord record) {
+        logger.info("Processing record: {}", record);
+
+        var oldCustomerOpt = convertToCustomerDto(record.getDynamodb().getOldImage());
+        var newCustomerOpt = convertToCustomerDto(record.getDynamodb().getNewImage());
+
+        if (oldCustomerOpt.isEmpty() && newCustomerOpt.isEmpty()) {
+            return List.of();
+        }
+
+        var customer = newCustomerOpt.or(() -> oldCustomerOpt).orElseThrow();
+        var customerId = customer.getId();
+        var organizationId = customer.getCristinId();
+
+        var oldClaims = oldCustomerOpt.map(CustomerDto::getChannelClaims).orElse(List.of());
+        var newClaims = newCustomerOpt.map(CustomerDto::getChannelClaims).orElse(List.of());
+
+        return computeChannelClaimEvents(customerId, organizationId, oldClaims, newClaims);
+    }
+
+    private String claimKey(ChannelClaimDto claim) {
+        return claim.channel().toString(); // Adjust fields as needed
+    }
+
+    private boolean isModified(ChannelClaimDto oldClaim, ChannelClaimDto newClaim) {
+        return !Objects.equals(oldClaim, newClaim);
+    }
+
+    private List<ResourceUpdateEvent<ChannelClaim>> computeChannelClaimEvents(URI customerId,
+                                                                                 URI organizationId,
+                                                                                 Collection<ChannelClaimDto> oldClaims,
+                                                                                 Collection<ChannelClaimDto> newClaims) {
+        var oldMap = oldClaims.stream()
+                         .collect(Collectors.toMap(this::claimKey, Function.identity()));
+
+        var newMap = newClaims.stream()
+                         .collect(Collectors.toMap(this::claimKey, Function.identity()));
```

Comment:
Do we need claimKey method here? Can we use ChannelClaimDto::channel and use channel URI as key in the map?

Reply:
Removed